submitit/test_snapshot.py

```diff
@@ -0,0 +1,13 @@
+import os
+from tempfile import TemporaryDirectory
+
+from .snapshot import SnapshotManager
+
+
+def test_snapshot():
+    cwd = os.getcwd()
+    with TemporaryDirectory() as tdir:
+        with SnapshotManager(tdir):
+            assert os.getcwd() == tdir
+            assert os.path.exists(os.path.join(tdir, "submitit/test_snapshot.py"))
+    assert os.getcwd() == cwd
```

Comment:
It could use more test like:

```
with SnapshotManager(snapshot_dir=tmp_path):
    job = executor.submit(os.getcwd)
    assert Path(job.result()) == tmp_path
```